### PR TITLE
Fix subresource discovery and versioning

### PR DESCRIPTION
**Pull request by liggitt**
Fixes https://github.com/kubernetes/kubernetes/issues/54684
Related to https://github.com/kubernetes/kubernetes/pull/54586

Allows distinct subresource group/version/kind to be used for each version (gives us a path to move to autoscaling/v1 for apps, or policy/v1 for eviction, etc)

Added tests to ensure scale subresources have expected discovery info, and that the object returned matches discovery, and that the endpoint accepts the advertised version

```release-note
Fixes discovery information for scale subresources in the apps API group
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 1]
```diff
@@ -162,12 +163,6 @@ func (a *APIInstaller) getResourceKind(path string, storage rest.Storage) (schem
 			fqKindToRegister = a.group.GroupVersion.WithKind(fqKind.Kind)
 			break
 		}
-
-		// TODO: keep rid of extensions api group dependency here
-		// This keeps it doing what it was doing before, but it doesn't feel right.
-		if fqKind.Group == "extensions" && fqKind.Kind == "ThirdPartyResourceData" {
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: I bet that felt good.

**Comment by liggitt** [comment]
Added integration test to ensure discovery matches the endpoint

**Pull request merged**